File: BubbleSort/BubbleSort_Implement/BubbleSort.java
package BubbleSort_Implement;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class BubbleSort {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());

		int[] data = new int[n];
		StringTokenizer st = new StringTokenizer(br.readLine());
		for (int i = 0; i < n; i++) {
			data[i] = Integer.parseInt(st.nextToken());
		}

		int temp = 0;

		/*
		 * 처음 짠 로직
		 * 
		 * for(int i = 0; i<n; i++) {//k번 반복하고 그 상태 그대로 출력.
		 * for(int j = 0; j<n-1-i; j++) {
		 * if(data[j]>data[j+1]) {
		 * temp = data[j];
		 * data[j] = data[j+1];
		 * data[j+1] = temp;
		 * }
		 * }
		 * }
		 * 
		 */

		for (int i = n; i > 0; i--) {
			for (int j = 0; j < i - 1; j++) {
				if (data[j] > data[j + 1]) {
					temp = data[j];
					data[j] = data[j + 1];
					data[j + 1] = temp;
				}
			}
		}

		System.out.println(Arrays.toString(data));

		// for(int i = 0; i<n; i++) {
		// System.out.printf("%d ", data[i]);
		// }
	}
}

File: BubbleSort/BubbleSort_Implement/BubbleSort.c
#include <stdio.h>

int main(){

    int n = 0;
    scanf("%d", &n);
    
    int data[n]; //n개의 값을 가지는 배열 선언
    for(int i=0; i<n; i++){
        scanf("%d", &data[i]);
    }
    
    int temp = 0;//임시로 데이터 저장할 변수 선언

    for(int i = 0; i < n-1; i++){//n-1만큼 반복
    	for(int j = 0; j< n-1-i; j++){//n-1만큼 비교하다 마지막 반복엔 1번 비교
    		if(data[j] > data[j+1]){
    			temp = data[j]; //data[j]를 잠시 저장
    			data[j] = data[j+1]; //data[j+1]을 data[j]의 위치로 이동
    			data[j+1] = temp;	//data[j+1]의 자리에 저장해둔 data[j]를 입력.
    			//큰 숫자가 뒤로 밀리는 오름차순 완성
    		}
    	}
    }
    
    for(int i=0; i<n; i++){
        printf("%d ", data[i]);
    }
    
    return 0;
}